### PR TITLE
Wren Hot Reloading

### PR DESCRIPTION
**Pull request by santiberna**
### Description

This pull request add the Inspector Module, the functionality to reset the scripting module and reload the main script and a button to do this on the main toolbar.

The inspector module separates most of the imgui functionality into its own module.
Wren VM hot reloading is relatively fast.

### Issues

[Codecks Issue](https://bubonic-brotherhood.codecks.io/card/1c3-wren-game-main-entry-function)

### Test criteria

You can test the Hot Reloading by playing with the initial starting position of the player in wren. It should also be tested if ImGui initialization and shutdown is handled correctly in the new module

### CODE REVIEW

File: engine/inspector/private/inspector_module.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if (auto* ptr = engine.GetModuleSafe<RendererModule>())
+    {
+        ptr->GetRenderer()->FlushCommands();
+    }
+    _editor.reset();
+}
+
+void InspectorModule::Tick(Engine& engine)
+{
+    _imguiBackend->NewFrame();
```
Comment (luuk234) [inline engine/inspector/private/inspector_module.cpp:54]: i would move all the other editor logic to here as well probably

**Review by luuk234 — APPROVED** [review]

File: engine/inspector/private/inspector_module.cpp (hunk 1 above)
Comment (santiberna) [reply engine/inspector/private/inspector_module.cpp:54]: I believe I got all of the logic, what am I missing something?

File: engine/include/old_engine.hpp [hunk 2]
```diff
@@ -27,13 +26,9 @@ class OldEngine : public ModuleInterface
     // std::unique_ptr<ThreadPool> _threadPool;
     // std::unique_ptr<AssetManager> _AssetManager;
 
-    std::unique_ptr<Editor> _editor;
     ECSModule* _ecs;
     glm::ivec2 _lastMousePos {};
     std::chrono::time_point<std::chrono::high_resolution_clock> _lastFrameTime;
-    PerformanceTracker _performanceTracker;
-
-    std::unique_ptr<PhysicsModule> _physicsModule;
```
[the remark this answers is not in the document]
Comment (santiberna) [reply engine/include/old_engine.hpp:32]: will check

File: engine/inspector/private/inspector_module.cpp (hunk 1 above)
Comment (luuk234) [reply engine/inspector/private/inspector_module.cpp:54]: well the editor draw is basically what draws all our imgui windows right, bit of an odd distinction to put the imgui titlebar in here but the rest of the editor logic in said function

Comment (santiberna) [reply engine/inspector/private/inspector_module.cpp:54]: Oh yeah you are right... I just thought moving all the editor code outside of that class could take some time and make the inspector module. I will add a TODO here

Comment (luuk234) [reply engine/inspector/private/inspector_module.cpp:54]: alright good enough, i also mentioned for the future since this is out of scope for this pr

**Pull request merged**